pkg/kubelet/envvars/envvars.go

```diff
@@ -34,16 +33,14 @@ func FromServices(services []*v1.Service) []v1.EnvVar {
 	for i := range services {
 		service := services[i]
 
-		// ignore services where ClusterIP is "None" or empty
-		// the services passed to this method should be pre-filtered
-		// only services that have the cluster IP set should be included here
-		if !v1helper.IsServiceIPSet(service) {
-			continue
-		}
-
 		// Host
 		name := makeEnvVariableName(service.Name) + "_SERVICE_HOST"
-		result = append(result, v1.EnvVar{Name: name, Value: service.Spec.ClusterIP})
+		if service.Spec.Type == v1.ServiceTypeExternalName {
+			result = append(result, v1.EnvVar{Name: name, Value: service.Spec.ExternalName})
```

Comment:
Right now, a ExternalName service cannot have an ClusterIP.

```golang
// clusterIP is the IP address of the service and is usually assigned
// randomly by the master. If an address is specified manually and is not in
// use by others, it will be allocated to the service; otherwise, creation
// of the service will fail. This field can not be changed through updates.
// Valid values are "None", empty string (""), or a valid IP address. "None"
// can be specified for headless services when proxying is not required.
// Only applies to types ClusterIP, NodePort, and LoadBalancer. Ignored if
// type is ExternalName.
// More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
// +optional
ClusterIP string `json:"clusterIP,omitempty" protobuf:"bytes,3,opt,name=clusterIP"`
```

Theres also a validation in place to ensure this.
`The Service "my-rds" is invalid: spec.clusterIP: Invalid value: "10.96.0.67": must be empty for ExternalName services`